From a85ae3b434565a3a33c4f82d9f62c303d9251499 Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Wed, 30 Sep 2020 17:58:45 +0530
Subject: [PATCH 1687/1852] pinctrl: zynqmp: Initialize the ret variable with
 zero

commit 0eb2933f98c3abcb338e4c8a099c4a20ef3fc721 from
https://github.com/Xilinx/linux-xlnx.git

Fix the possibility of retuning garbage value by initializing the
ret variable with zero. One of the return paths in the function
returning uninitialized ret variable.

Addresses-coverity: Uninitialized scalar variable (UNINIT).
Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pinctrl/pinctrl-zynqmp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-zynqmp.c b/drivers/pinctrl/pinctrl-zynqmp.c
index 8b7d44b32461..50cb4c8ba5e5 100644
--- a/drivers/pinctrl/pinctrl-zynqmp.c
+++ b/drivers/pinctrl/pinctrl-zynqmp.c
@@ -670,7 +670,7 @@ static int zynqmp_pinctrl_prepare_func_groups(struct device *dev, u32 fid,
 {
 	u16 resp[NUM_GROUPS_PER_RESP] = {0};
 	const char **fgroups;
-	int ret, index, i;
+	int ret = 0, index, i;
 
 	fgroups = devm_kzalloc(dev, sizeof(*fgroups) * func->ngroups,
 			       GFP_KERNEL);
@@ -852,7 +852,7 @@ static int zynqmp_pinctrl_prepare_group_pins(struct device *dev,
 					     unsigned int ngroups)
 {
 	unsigned int pin;
-	int ret;
+	int ret = 0;
 
 	for (pin = 0; pin < zynqmp_desc.npins; pin++) {
 		ret = zynqmp_pinctrl_create_pin_groups(dev, groups, pin);
-- 
2.31.1

